pkg/spire/controller.go

```diff
@@ -45,14 +47,30 @@ func init() {
 // controllerKey is a way to associate the ControllerAPIClient from inside the context.Context
 type controllerKey struct{}
 
+// OnStore stores the changed spire config into the SpireClientApi
+func OnStore(ctx context.Context, logger *zap.SugaredLogger) func(name string, value interface{}) {
+	return func(name string, value interface{}) {
+		if name == config.GetSpireConfigName() {
+			cfg, ok := value.(*spireconfig.SpireConfig)
+			if !ok {
+				logger.Error("Failed to do type assertion for extracting SPIRE config")
+				return
+			}
+			controllerAPIClient := GetControllerAPIClient(ctx)
+			controllerAPIClient.Close()
+			controllerAPIClient.SetConfig(*cfg)
```

Comment:
Should the new values be validated before updating the config map like we did [here](https://github.com/tektoncd/pipeline/blob/77c169876707295bc7dcb7b3161bd2278564a5cd/pkg/apis/config/spire_config.go#L66)?

Reply:
OnStore is a callback called by Store or UntypedStore which watches the configs and if changed calls NewSpireConfigFromConfigMap which in turn does the validation.  So by the point we come here the validation is already done.

Reply:
sgtm.